src/utilities.jl

```diff
@@ -115,3 +115,61 @@ end  # setcolnames!
 
     true
 end
+
+# helper method for inner constructor
+@inline function _issorted_and_unique(x)
+    for i in 1:length(x)-1
+        @inbounds !(x[i] < x[i + 1]) && return false
+    end
+    true
+end
+
+# helper method for inner constructor
+function replace_dupes(cnames::Vector{Symbol})
+    n = 1
+    while !allunique(cnames)
+        ds = find_dupes_index(cnames)
+        for d in ds
+            if n == 1
+                cnames[d] = Symbol(cnames[d], "_$n")
+            else
+                cnames[d] = Symbol(cnames[d][1:length(cnames[d])-length(string(n))-1], "_$n")
+            end
+        end
+        n += 1
+    end
+    cnames
+end
+
+# helper method for inner constructor
+find_dupes_index(A::Vector{Symbol}) =
+    @inbounds [i for i in eachindex(A) if A[i] ∈ A[1:i-1]]
+
+@memoize function gen_colnames(n::Integer)
```

Comment:
Have you seen a notable performance improvement when using `@memoize`?

Reply:
yeah, this function is quite slow. (or my algo is bad, is there better algo?

Orig:
```
julia> @btime TimeSeries.gen_colnames(5)
  899.902 ns (9 allocations: 384 bytes) 
5-element Array{Symbol,1}:
 :A
 :B
 :C
 :D
 :E
```
vs `@memoize`
```
julia> @btime TimeSeries.gen_colnames(5)
  92.643 ns (2 allocations: 32 bytes)
5-element Array{Symbol,1}:
 :A
 :B
 :C
 :D
 :E
```

Reply:
Interesting. I'll take a look at the algorithm itself in more detail later, this is fine in the meantime I suppose.

Reply:
IIRC it's just `x` appended with sequential numbers, i.e. `x1`, `x2`, ...